beps/0001-notifications-system/README.md

```diff
@@ -85,38 +87,44 @@ The role of the notifications plugin is to manage the lifecycle of notifications
 The notification backend stores notification using the [database service](https://backstage.io/docs/backend-system/core-services/index#database). In particular it needs to store the following information for each notification:
 
 - ID
-- Title
-- Priority
-- Origin
-- Topic (optional)
-- Creation timestamp (generated)
-- Receivers
-- Read status by user
-- Payload
-  - icon (optional)
-  - message (optional)
-  - links (list, optional)
-
-The receivers is **not** a list of users, but rather a filter that describes who should receive the notification. It must be possible to evaluate this filter in a database query, so that we can efficiently fetch all notifications for a given user. The same filter will also be used by the signal backend to determine which users should receive a signal.
-
-The title is mandatory human readable text shortly describing the notifications.
-
-The priority is one of the predefined values to help with presentation of the notifications or their filtering. Values: Critical, High, Normal, Low.
+- Read date
+- Done date
+- Saved status
+- Creation date
+- Updated date (optional, for topic notifications)
+- Payload:
+  - Title
+  - Description
+  - Priority
+  - Origin
+  - Link
+  - Recipients
+  - Topic (optional)
+  - Icon (optional)
+  - Image URL (optional)
+
+The recipients is **not** a list of users, but rather a filter that describes who should receive the notification. It must be possible to evaluate this filter in a database query, so that we can efficiently fetch all notifications for a given user. The same filter will also be used by the signal backend to determine which users should receive a signal.
+
+The title is mandatory human-readable text shortly describing the notifications.
+
+The description is an optional human-readable text providing more details to the user.
+
+The priority is one of the predefined values to help with presentation of the notifications or their filtering. Allowed values are: `critical`, `high`, `normal` (default), and `low`.
 
 The origin is a string identifying the creator of the notification, i.e. the BE plugin or external system.
 
-The topic is an optional string helping to group notifications of particular context. Its use cases include:
+If a topic is provided for a notification, it will either:
 
-- several notifications emitted by an asynchronous external task can be grouped by a single topic
-- a BE plugin can group several related messages to a particular processing, i.e. asynchronous progress monitoring
+- Create a new notification if user doesn't have existing notification in that topic
+- Update existing notification in that topic so that it's unread considered as a new notification
 
-The timestamp of message creation is auto-generated by the notifications backend at the time of receiving a request to create the notification.
+The idea of topic is to be able to reuse notifications about same information for example specific failing CI build.
```

Comment:
This is a good use-case where we do not need/want to keep history.

But not the only one. The topic is meant to group messages. The user might be interested in keeping their history - an async task can consist of multiple smaller steps which result is the user interested in (timing, logs, etc).
Example with Onboard a new user:
 - Create email account (info with details about the it)
 - Create access to a system (link with a secret or logs)
 - create user space in an XYZ system (lwith a link to it)
 - ... 

I think it's about consumer (FE plugin) to decide whether present or skip the history.
The Notifications Page will list whatever is there.